pkg/controllers/cluster/import.go

```diff
@@ -233,3 +235,27 @@ func (i *importHandler) importCluster(cluster *fleet.Cluster, status fleet.Clust
 	status.Agent = fleet.AgentStatus{}
 	return status, nil
 }
+
+func (i *importHandler) restConfigFromKubeConfig(data []byte) (*rest.Config, error) {
+	clientConfig, err := clientcmd.NewClientConfigFromBytes(data)
+	if err != nil {
+		return nil, err
+	}
+
+	raw, err := clientConfig.RawConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	if raw.Contexts[raw.CurrentContext] != nil {
+		cluster := raw.Contexts[raw.CurrentContext].Cluster
+		if raw.Clusters[cluster] != nil {
+			_, err := http.Get(raw.Clusters[cluster].Server)
+			if err == nil {
+				raw.Clusters[cluster].CertificateAuthorityData = nil
+			}
+		}
+	}
```

Comment:
We may want to return an error if `raw.Contexts[raw.CurrentContext]` is `nil`.

Reply:
We might not want to handle validation at here. This piece of code is to check if the current cluster is behind a public CA and if so remove the cacert data in kubeconfig. So if there is no cluster it will fail to establish a contection and other places will return an error.

Reply:
Ah, that makes sense given the context.